### PR TITLE
fix: (Platform) Add outline on nodes, detects window change for Approval flow component

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
#4225 #4235 #4236 

#### Please provide a brief summary of this pull request.
This PR fixes issues linked above.
1. Detect window resize and show navigation button
2. fix focus outline for avatar and approval nodes
3. fix accessibility issue

1. #4235 issue for stackblitz is not handled in this PR, as that is a general issue for all components like Core calendar example for special days. This needs to be investigated and fixed in a different PR. 
Here stackblitz is appending .component in extra data class files provided. so examples not working on stackblitz.

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] Documentation Examples

### CODE REVIEW

**Comment by DeepakSap14** [comment]
Screenshot 2021-01-11 at 5 53 59 PM

File: libs/platform/src/lib/components/approval-flow/approval-flow.component.ts [hunk 1]
```diff
@@ -90,6 +90,7 @@ export class ApprovalFlowComponent implements OnInit, OnDestroy {
     _dir: string;
 
     private subscriptions = new Subscription();
+    private _onDestroy$: Subject<void> = new Subject<void>();
```
Comment (kavya-b) [inline libs/platform/src/lib/components/approval-flow/approval-flow.component.ts:93]: call in onDestroy too

File: libs/platform/src/lib/components/approval-flow/approval-flow.component.scss [hunk 2]
```diff
@@ -18,6 +18,13 @@ $block: approval-flow;
         cursor: pointer;
     }
 
+    fd-avatar:focus {
+        outline-offset: 0.0625rem;
+        outline-width: 0.0625rem;
+        outline-style: dotted;
+        outline-color: #0854A0;
```
Comment (IlyaSurmay) [inline libs/platform/src/lib/components/approval-flow/approval-flow.component.scss:25]: please add `outline-color: var(--sapContent_FocusColor, #0854A0); ` here and in other places in this file

Comment (DeepakSap14) [reply libs/platform/src/lib/components/approval-flow/approval-flow.component.scss]: fixed

File: libs/platform/src/lib/components/approval-flow/approval-flow.component.ts (hunk 1 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/approval-flow/approval-flow.component.ts:93]: fixed

File: libs/platform/src/lib/components/approval-flow/approval-flow.component.ts [hunk 3]
```diff
@@ -261,6 +264,17 @@ export class ApprovalFlowComponent implements OnInit, OnDestroy {
     /** @hidden */
     ngOnDestroy(): void {
         this.subscriptions.unsubscribe();
+        this._onDestroy$.complete();
```
Comment (IlyaSurmay) [inline libs/platform/src/lib/components/approval-flow/approval-flow.component.ts:267]: is this going to work without `this._onDestroy$.next();`?

Comment (IlyaSurmay) [reply libs/platform/src/lib/components/approval-flow/approval-flow.component.ts:267]: also it'd be better not to mix approaches for unsubs and pick only one, now `_onDestroy ` is used along with `this.subscriptions`

File: libs/platform/src/lib/components/approval-flow/approval-flow.component.html [hunk 4]
```diff
@@ -17,15 +17,13 @@ <h3 class="approval-flow__title" [attr.dir]="_dir">{{ title }}</h3>
...
+        <button class="approval-flow__control--prev-slide" *ngIf="_carouselStep > 0" (click)="previousSlide()"
+            aria-label="Go to previous slide">
             <fd-icon [glyph]="'navigation-' + (_isRTL ? 'right' : 'left') + '-arrow'"></fd-icon>
             {{ _carouselStep }}
         </button>
-        <button class="approval-flow__control--next-slide"
-                *ngIf="(_maxCarouselStep - _carouselStep) > 0"
-                (click)="nextSlide()">
+        <button class="approval-flow__control--next-slide" *ngIf="(_maxCarouselStep - _carouselStep) > 0"
+            (click)="nextSlide()" aria-label="Go to next slide">
```
Comment (olshansky) [inline libs/platform/src/lib/components/approval-flow/approval-flow.component.html:26]: Most probably this `aria-label` should have `i18n`

File: libs/platform/src/lib/components/approval-flow/approval-flow.component.ts [hunk 5]
```diff
@@ -261,6 +264,17 @@ export class ApprovalFlowComponent implements OnInit, OnDestroy {
...
+    }
+
+    /** @hidden Listen window resize and distribute cards on column change */
+    private _listenOnResize(): void {
+        fromEvent(window, 'resize')
+            .pipe(debounceTime(60), takeUntil(this._onDestroy$))
+            .subscribe(() => {
+                this._resetCarousel();
+                this._checkCarouselStatus();
+            });
```
Comment (olshansky) [inline libs/platform/src/lib/components/approval-flow/approval-flow.component.ts:277]: Maybe it should be added in `this.subscriptions ` for unsubscribing in `ngOnDestroy`

**Review by olshansky — APPROVED** [review]

**Review by sKudum — APPROVED** [review]

File: libs/platform/src/lib/components/approval-flow/approval-flow.component.ts (hunk 5 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/approval-flow/approval-flow.component.ts]: added in this.subscriptions

File: libs/platform/src/lib/components/approval-flow/approval-flow.component.html (hunk 4 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/approval-flow/approval-flow.component.html]: added i18n-aria-label

File: libs/platform/src/lib/components/approval-flow/approval-flow.component.ts (hunk 3 above)
Comment (DeepakSap14) [reply libs/platform/src/lib/components/approval-flow/approval-flow.component.ts]: using this.subscriptions now.

File: libs/platform/src/lib/components/approval-flow/approval-flow.component.ts [hunk 6]
```diff
@@ -90,6 +91,7 @@ export class ApprovalFlowComponent implements OnInit, OnDestroy {
     _dir: string;
 
     private subscriptions = new Subscription();
+    private _onDestroy$: Subject<void> = new Subject<void>();
```
Comment (IlyaSurmay) [inline libs/platform/src/lib/components/approval-flow/approval-flow.component.ts:94]: please remove it since it's not used anymore

Comment (DeepakSap14) [reply libs/platform/src/lib/components/approval-flow/approval-flow.component.ts]: done

**Review by IlyaSurmay — APPROVED** [review]

**Pull request merged**